src/checker.rs

```diff
@@ -0,0 +1,117 @@
+use std::error::Error;
+
+use reqwest::{Response, StatusCode};
+
+#[derive(PartialEq, Eq, PartialOrd, Ord, Hash, Debug)]
+pub enum CheckResult {
+    SUCCESS,
+    FAILURE(FailureReason),
+    MISSED,
+}
+
+#[derive(PartialEq, Eq, PartialOrd, Ord, Hash, Debug)]
+pub enum FailureReason {
+    Timeout,
+    DnsError(String),
+    Error(String),
+}
+
+pub async fn do_request(
+    client: &reqwest::Client,
+    domain: String,
+) -> Result<Response, reqwest::Error> {
+    /*
+    Fetches the response from a URL. First attempts to fetch just the head, and if not supported falls
+    back to fetching the entire body.
+     */
```

Comment:
If these are intended as doc comments, they should go above the function with triple slashes, like so:
```suggestion
/// Fetches the response from a URL.
///
/// First attempts to fetch just the head, and if not supported falls
/// back to fetching the entire body.
pub async fn do_request(
    client: &reqwest::Client,
    domain: String,
) -> Result<Response, reqwest::Error> {
```